### PR TITLE
use style sheets

### PR DESCRIPTION
**Pull request by rafaqz**
Sorry this took so long. It's pretty simple really, just moving the hard coded styling out to style.css, and tweaking things a little.

I swapped out the % based layout for full caption text and flexgrow for the _slider_. The right hand number is fixed em for regular slider (so it doesn't jump around when the text size changes) and zero width for the rangeslider.

It seems that the rangeslider is a little different with regards to passing in attributes etc? I have been passing in hover text, and that doesn't seem possible with the rangeslider? noui issue?

### CODE REVIEW

File: src/classes.jl [hunk 1]
```diff
@@ -8,6 +8,8 @@ function getclass(T::WidgetTheme, arg, typ...)
         typ[1]=="toggle" && return "switch"
         typ==("range",) && return "slider"
```
Comment (rafaqz) [inline src/classes.jl]: Having range and rangeslider is a bit confusing now, can we change this to slider?

**Comment by piever** [comment]
Thanks a lot! I'll be traveling from tomorrow till next Sunday but will review when I'm back. I wanted to ask: does this also fix https://github.com/JuliaGizmos/Interact.jl/issues/248 (see also #89 ) ? It'd be ideal if the same solution could fix both sliders and togglebuttons

File: src/slider.jl [hunk 2]
```diff
@@ -153,7 +153,9 @@ function rangeslider(::WidgetTheme, vals::AbstractRange{<:Integer}, formatted_va
         $fromJS[] = false
     end)
 
+    class = getclass(:input, "rangeslider", orientation)
     style = Dict{String, Any}(string(key) => val for (key, val) in style)
+
     haskey(style, "flex-grow") || (style["flex-grow"] = "1")
     !haskey(style, "height") && orientation == "vertical" && (style["height"] = "20em")
```
Comment (rafaqz) [inline src/slider.jl]: These lines are more hard-coded css that could be avoided by adding rangeslider-vertical class at this level and some css. But it seems not to appear if you add a className the scp.dom node. I don't totally get how this works and interacts with noui slider yet. Maybe we need another wrapper diva around the noui slider.

**Comment by rafaqz** [comment]
I doub tit will solve that, but I don't use jupyter anymore so I can't easily check. These PRs also clash almost 100%!

The main idea here was to get these changes happening in css generally so all of this will be easier to fix and change later, but it wont actually fix much, except that the sliders both look a lot better.

**Comment by piever** [comment]
Most of the hardcoded inline CSS is me hacking around without really understanding flexbox :) it'd be great to get rid of it. Ideally passing some CSS in a call to `slider(...; style=Dict(...))` should also work and we need to make sure it interacts nicely with or defaults. Currently (on master) this is not the case: if we pass a custom width the slider value is no longer next to the slider (say `slider(1:100, style=Dict("width" => "200px"))`). It's probably good to play around here to make sure we don't get this kind of issues.

File: src/slider.jl (hunk 2 above)
Comment (piever) [reply src/slider.jl]: The nouislider docs explain a bit how to customize the slider. Adding a wrapper div is also fine: the current implementation is definitely a bit hacky. Here also we'd want to allow users to pass custom className to the relevant node but I'm not sure this has been carefully implemented yet.

Comment (rafaqz) [reply src/slider.jl]: Also style and attributes should be easy to pass to the wrapper div

**Comment by rafaqz** [comment]
So the general goals are (?):
1. css overrides work without using `!important` on everything. This means everything needs an easily targeted id or class.
2. programmatic style and class overrides are also possible, which will also override css styles from 1
3. the process for applying css and style overrides is consistent and logical for all widgets

**Comment by djsegal** [comment]
Can I ask why there isn't a simple version of interact for run of the mill jupyter usage?

It seems a little ridiculous you broke a good tool so you can make apps in whatever you're using?

**Comment by rafaqz** [comment]
@djsegal this is a very specific pull request about using classes and css, and you seem to misunderstand the purpose or what that means.

This change makes Interact more simple and standard, not more complicated. I would also like to incorporate your fixes to Jupyter styling, and this is the right way to do it: with the standard web design methodology - classes and css. Hit F12 on your browser, click the cursor icon in the left hand corner, and take a look around how they built this page. Classes and ids, with style sheets. You will struggle to find more than a handful of `style` tags, and there are some very good reasons for that. 

And btw, run of the mill usage for me is electron/mux apps, which I use everyday and share with my colleages. Jupyter is pretty bad for making interfaces for models of any complexity.

File: src/classes.jl (hunk 1 above)
Comment (piever) [reply src/classes.jl]: I'm not even sure I want to keep `getclass` to be honest... It was initially used to allow different CSS framework to have different class names but I don't think it's very relevant now as the consensus seemed to be to only have one CSS framework and allow the option to users to write their own CSS with these class names, so I wouldn't worry about that for now.

**Comment by piever** [comment]
Looks good! I'm merging this as is as it establishes the correct structure to add style. There are I think three things left to do (but can be done in separate PRs):

1. Figure out the noUIslider case to avoid the hardcoded style hack
2. Fix the custom width to the slider: if I specify the width of the slider, the flexbox still grows and one gets unwanted space, for example:

screenshot from 2018-10-01 12-52-43

3. Port the fix in #89 to this new approach

File: src/classes.jl (hunk 1 above)
Comment (rafaqz) [reply src/classes.jl]: I agree, it could be unnecessary modularity at this stage.

**Comment by piever** [comment]
Regarding the Jupyter digression, while most people seem to use this with Jupyter, we should probably add a word of caution that things tend to work better outside of it as Jupyter tries to enforce it's own stlye to the inline output. Electron app or web app with Mux on localhost tend to work better, but this discussion should be had on a separate issue.

**Pull request merged**